capstone-rs/src/instruction.rs

```diff
@@ -157,12 +158,64 @@ impl<'a> Drop for Instructions<'a> {
     }
 }
 
+/// Shared functionality for structs wrapping [`cs_insn`]
+///
+/// # Safety
+///
+/// The [`cs_insn::detail`] pointer must be safely allocated and correspondingly freed
+/// to implement this trait safely.
+#[allow(clippy::len_without_is_empty)]
+pub unsafe trait Instruction {
```

Comment:
Since you already `impl<'a> Deref for OwnedInsn<'a>`, is the trait necessary? It exposes methods that we don't want public like `insn()` and `detail()`.

There is no public trait in `std` for all `str` methods; `String` just derefs to `str`.